src/app/loadout-builder/generated-sets/GeneratedSets.tsx

```diff
@@ -10,15 +11,64 @@ import {
   armor2PlugCategoryHashesByName,
 } from 'app/search/d2-known-values';
 import _ from 'lodash';
-import React, { Dispatch, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import React, { Dispatch, useCallback, useEffect, useRef, useState } from 'react';
 import { List, WindowScroller } from 'react-virtualized';
 import { DimStore } from '../../inventory/store-types';
 import { LoadoutBuilderAction } from '../loadoutBuilderReducer';
+import { someModHasEnergyRequirement } from '../mod-utils';
 import { ArmorSet, LockedMap, LockedModMap, StatTypes } from '../types';
-import { someModHasEnergyRequirement } from '../utils';
 import GeneratedSet from './GeneratedSet';
 import styles from './GeneratedSets.m.scss';
 
+/** Taller item groups have either the swap icon under the item of an exotic perk. */
+function hasExoticPerkOrSwapIcon(items: DimItem[]) {
+  return items.length > 1 || items.some((item) => item.isExotic);
+}
+
+/**
+ * Gets the set used to measure how high a row can be. It also returns a recalcTrigger that is
+ * intended to trigger a recalculation when the value changes.
+ *
+ * It figures out the tallest row by looking at items with exotic perks and swap icons.
+ * Exotic perks add another row to the mod icons and the swap icon sits below the item image.
+ * The height they add is roughly equivalent so we treat both conditions equally.
+ */
+function getMeasureSetAndIndicator(sets, isPhonePortrait): [ArmorSet | undefined, number] {
```

Comment:
This fixes issues where we could see the swap icon falling below the bottom border. Also mobile rows are correctly sized now. I was having issues with my new buttons sitting over the next row. This is all fixed now